### ISSUE TITLE
Built-in Git Sync error: Host key verification failed. fatal: Could not read from remote repository. Please make sure you have the correct access rights and the repository exists.

### ISSUE THREAD

**Issue by phifa** [Microsoft/vscode#56664]
After installing the new non-admin VS Code version, Git Sync Button at the bottom left is not working properly anymore. Git Log says: 

```
> git pull --tags origin develop
Host key verification failed.
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
```

When I start VS Code with "run as Admin", it works.

**Comment by joaomoreno** [comment]
This is very odd. Which OS are you on?

**Comment by phifa** [comment]
I just had a similar issue: I clicked on the sync button (running VSC as Administrator), got an error: 
Git: Permission denied (publickkey). It suggest to look at the git log: 

```
> git fetch
> git pull --tags origin master
Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
Permission denied (publickey).
fatal: Could not read from remote repository.
```

Now when I manually run in my integrated WSL terminal: `git fetch` or `git pull --tags origin master` everything works, and my sync button updates in the background, showing no differences between local and origin repository.

**Comment by phifa** [comment]
Am I supposed to run WSL and VS Code as Administrator or not?

**Comment by joaomoreno** [comment]
You are supposed to **never** run VS Code as Administrator. There's no reason to do that. Once you do it, VS Code might create files which only an Administrator can access, so the result is that you can't even use VS Code without running as Administrator again.

That seems exactly to be what has happened: You set up your SSH using an Admin command line prompt. Please setup your SSH auth again, but this time for your user.

**Comment by phifa** [comment]
Ok. That’s good to know. I think the reason I did this at some point was because of the integrated terminal not having enough permission to create symlinks or do sth else, not sure. This was when I was using the git bash. Now I have included the WSL. 

How does running VS Code as Admin change the behavior of the integrated terminal, the WSL?

**Comment by phifa** [comment]
So here is the test i did: 

- I made sure Hyper/WSL is not starting in Admin mode. I then cloned my private (!) repository again with Hyper/WSL. Worked fine. 
- I then did `code projectname` and made sure VS Code does not run as Admin (as I can easily doublecheck in the titlebar, great). 
- In VS Code clicking on the sync button, brought the error again. When I open the integrated terminal I could do `git push` and `git pull` without errors. Also pushing and pulling works via Hyper/WSL just fine. 

So the problem persists, but only when I use the built-in git tools. How do the built-in git tools authorize? How can I setup my SSH auth again, as you described?
